velox/exec/tests/utils/PlanBuilder.cpp

```diff
@@ -899,6 +899,57 @@ PlanBuilder& PlanBuilder::localMerge(const std::vector<std::string>& keys) {
   return *this;
 }
 
+PlanBuilder& PlanBuilder::expand(
+    const std::vector<std::vector<std::string>>& projections) {
+  std::vector<std::vector<core::TypedExprPtr>> projectExprs;
+  projectExprs.reserve(projections.size());
+  const auto numColumns = projections[0].size();
+  const auto numRows = projections.size();
+  std::vector<std::string> names;
+  names.reserve(numColumns);
+  std::vector<TypePtr> types;
+  types.reserve(numColumns);
+  int grouIdColCount = 0;
+  const auto inputRowType = planNode_->outputType();
+
+  for (auto i = 0; i < numRows; i++) {
+    std::vector<core::TypedExprPtr> projectExpr;
+    for (auto j = 0; j < numColumns; j++) {
+      auto untypeExpression = parse::parseExpr(projections[i][j], options_);
+      auto typeExpression = inferTypes(untypeExpression);
+
+      if (names.size() != numColumns) {
+        if (auto fieldExpr = dynamic_cast<const core::FieldAccessExpr*>(
+                untypeExpression.get())) {
+          names.push_back(fieldExpr->getFieldName());
+        } else if (untypeExpression->alias().has_value()) {
+          names.push_back(untypeExpression->alias().value());
+        }
+      }
+
+      auto fieldExpr =
+          dynamic_cast<const core::ConstantExpr*>(untypeExpression.get());
+      if (fieldExpr) {
```

Comment:
This logic seems strange? What is it for?

Reply:
@mbasmanova The logic here is to determine if the field is of const type. If it is, then check if it is null. If it is null, it means the field column is empty. Otherwise, it is the group ID column.